src/main/scala/org/constellation/primitives/Genesis.scala

```diff
@@ -2,81 +2,66 @@ package org.constellation.primitives
 import io.chrisdavenport.log4cats.slf4j.Slf4jLogger
-import io.chrisdavenport.log4cats.{Logger, SelfAwareStructuredLogger}
-import org.constellation.domain.consensus.ConsensusStatus
+import org.constellation.DAO
 import org.constellation.keytool.KeyUtils
 import org.constellation.primitives.Schema._
 import org.constellation.schema.Id
-import org.constellation.serializer.KryoSerializer
 import org.constellation.util.AccountBalance
-import org.constellation.{ConfigUtil, ConstellationExecutionContext, DAO}
-
-import scala.util.{Failure, Success, Try}
 
 object Genesis extends StrictLogging {
 
   implicit val unsafeLogger: SelfAwareStructuredLogger[IO] = Slf4jLogger.getLogger[IO]
 
-  final val CoinBaseHash = "coinbase"
-  final val DebtKey = KeyUtils.makeKeyPair()
+  final val Coinbase = "coinbase"
+  // TODO: That key should be provided externally by the genesis creator.
+  // We can use Node's KeyPair as well.
+  final val CoinbaseKey = KeyUtils.makeKeyPair()
 
   private final val GenesisTips = Seq(
-    TypedEdgeHash(CoinBaseHash, EdgeHashType.CheckpointHash),
-    TypedEdgeHash(CoinBaseHash, EdgeHashType.CheckpointHash)
+    TypedEdgeHash(Coinbase, EdgeHashType.CheckpointHash),
+    TypedEdgeHash(Coinbase, EdgeHashType.CheckpointHash)
   )
 
-  def createGenesisTransaction(keyPair: KeyPair, allocAccountBalances: Seq[AccountBalance])(
-    implicit dao: DAO
-  ): Seq[Transaction] =
-    (allocAccountBalances.map(
-      t => dao.transactionService.createTransaction(DebtKey.address, t.accountHash, t.balance, DebtKey)
-    ) :+
-      dao.transactionService
-        .createTransaction(DebtKey.address, keyPair.getPublic.toId.address, 4e9.toLong, keyPair)).toList.sequence.unsafeRunSync
-
-  def createGenesisBlock(keyPair: KeyPair, allocAccountBalances: Seq[AccountBalance])(
-    implicit dao: DAO
-  ): CheckpointBlock =
-    CheckpointBlock.createCheckpointBlock(createGenesisTransaction(keyPair, allocAccountBalances), GenesisTips)(keyPair)
-
-  def start()(implicit dao: DAO): Unit = {
-    // TODO: Remove initial distribution
-    val fakeIdToGenerateTips = KeyUtils.makeKeyPair().getPublic.toId
-    val go = createGenesisAndInitialDistributionDirect(
-      dao.selfAddressStr,
-      Set(fakeIdToGenerateTips),
-      dao.keyPair,
-      dao.nodeConfig.allocAccountBalances
-    )
-    acceptGenesis(go, setAsTips = true)
-  }
-
-  // TODO: Get rid of this, need to add edge case for handling tips at beginning to avoid this
-  def createDistribution(
-    selfAddressStr: String,
-    ids: Seq[Id],
-    genesisSOE: SignedObservationEdge,
-    keyPair: KeyPair
-  )(implicit dao: DAO): CheckpointBlock = {
+  def createDistributionTransactions[F[_]: Sync](
+    allocAccountBalances: Seq[AccountBalance]
+  )(implicit dao: DAO): F[Seq[Transaction]] =
+    allocAccountBalances.toList
+      .traverse(ab => dao.transactionService.createTransaction(Coinbase, ab.accountHash, ab.balance, CoinbaseKey))
+      .asInstanceOf[F[Seq[Transaction]]]
```

Comment:
Why it's needed? You have `F[List[Transaction]]` by the way

Reply:
Here the situation is tricky. I have already started making the Genesis async.

So we have this method which has implicit `F[_]: Sync` buuuut the `dao.transactionService` is `TransactionService[IO]` instead of `TransactionService[F]` and because of type variance I need to temporarily cast it to more generic type ;/ So basically it won't be needed when we refactor the Genesis. I will double check it though and eventually put the proper comment there

Reply:
Yep. Calling `toSeq` or changing the type to `F[List[Transaction]]` solves the issue but only partially.

The problem was with missing type parameter in `createTransaction` method, so:

```scala
class TransactionService[F[_]: Concurrent] {
   def createTransaction(...) { ... }
}
```

needs to be changed to:

```scala
class TransactionService[F[_]: Concurrent] {
   def createTransaction[F[_]: Concurrent](...) { ... }
}
```